### PR TITLE
planner, executor: group_concat(multiple cols) should not be converted to proj

### LINKED ISSUE #9922: wrong aggEliminateCheck for `select group_concat() group by unique`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
``` sql
create table t(id int primary key, a int, b int);
insert into t values(1,1,1);
insert into t values(2,1,1);
select group_concat(a,b) from t group by id;
```

2. What did you expect to see?
``` sql
+-------------------+
| group_concat(a,b) |
+-------------------+
| 11                |
| 11                |
+-------------------+
```

3. What did you see instead?
``` sql
+-------------------+
| group_concat(a,b) |
+-------------------+
| 1                 |
| 1                 |
+-------------------+
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
``` sql
Release Version: v3.0.0-beta.1-8-g7cabaac61
Git Commit Hash: 7cabaac61dbc7fa65948ee7fbd11018af7765369
Git Branch: master
UTC Build Time: 2019-03-27 03:25:00
GoVersion: go version go1.12 darwin/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve? 
fix #9922 

### What is changed and how it works?
Check whether an aggregation function is group_concat with multiple columns,
if so, do not eliminate it during aggregation eliminating.

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported function/method change

Side effects

 - Possible performance regression

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: planner/core/rule_aggregation_push_down.go [hunk 1]
```diff
@@ -299,91 +299,95 @@ func (a *aggregationPushDownSolver) pushAggCrossUnion(agg *LogicalAggregation, u
...
-						lChild = a.tryToPushDownAgg(leftAggFuncs, leftGbyCols, join, 0)
-					}
-					join.SetChildren(lChild, rChild)
-					join.SetSchema(expression.MergeSchema(lChild.Schema(), rChild.Schema()))
-					join.buildKeyInfo()
-					proj := a.tryToEliminateAggregation(agg)
-					if proj != nil {
-						p = proj
-					}
+		p, err = a.tryEliminateAgg(agg)
```
Comment (eurekaka) [inline planner/core/rule_aggregation_push_down.go]: `p` can be nil and used later?

Comment (winoros) [inline planner/core/rule_aggregation_push_down.go]: This error can be ignored?

**Comment by XuHuaiyu** [comment]
@eurekaka @winoros 
I'll split this PR into 2 PRs:
1. fix issue #9922 : #9967 
2. do refactor

**Comment by zz-jason** [comment]
@XuHuaiyu Can we close this PR?

**Pull request closed without merging**